py/circuitpy_mpconfig.h

```diff
@@ -179,17 +179,11 @@ typedef long mp_off_t;
 // board-specific definitions, which control and may override definitions below.
 #include "mpconfigboard.h"
 
-// CIRCUITPY_FULL_BUILD is defined in a *.mk file.
-
+// CIRCUITPY_FULL_BUILD is defined in py/circuitpy_mpconfig.mk
 // Remove some lesser-used functionality to make small builds fit.
 #define MICROPY_BUILTIN_METHOD_CHECK_SELF_ARG (CIRCUITPY_FULL_BUILD)
-//TODO: replace this with a rework of the FULL_BUILD system
-#if !defined(MICROPY_CPYTHON_COMPAT)
-	#define MICROPY_CPYTHON_COMPAT                (CIRCUITPY_FULL_BUILD)
-#endif
-#if !defined(MICROPY_COMP_FSTRING_LITERAL)
+#define MICROPY_CPYTHON_COMPAT                (CIRCUITPY_FULL_BUILD)
 #define MICROPY_COMP_FSTRING_LITERAL          (MICROPY_CPYTHON_COMPAT)
-#endif
```

Comment:
I didn't find any evidence that MICROPY_COMP_FSTRING_LITERAL was presently defined before reaching this line; the compiler should shout if it was.  If it's accepted, and if no boards suddenly overflow due to fstring literals being compiled in, it's probably fine.